src/Editor/SourceView.tsx

```diff
@@ -4,6 +4,37 @@ import { getCode } from "../JavascriptVM/vmSlice";
 import SyntaxHighlighter from "react-syntax-highlighter";
 
 import "./SourceView.css";
+import { BLOCKLY_HIGHLIGHT_PREFIX } from "../BlocklyInterface/BlocklyInstance";
+
+/**
+ * Removes any occurrences of the BLOCKLY_HIGHLIGHT_PREFIX in the
+ * provided code.
+ *
+ * @param blocklyCode Code that might contain BLOCKLY_HIGHLIGHT_PREFIX
+ * statements.
+ *
+ * @returns A string that is identical to the provided 'blocklyCode',
+ * but without BLOCKLY_HIGHLIGHT_PREFIX statements.
+ */
+function removeHighlightBlockCalls(blocklyCode: string | null): string {
+  if (!blocklyCode) return "";
+
+  const blocklyCodeLines: Array<string> = blocklyCode.split("\n");
+  let output: string = "";
+  for (let i = 0; i < blocklyCodeLines.length; ++i) {
```

Comment:
I'd use an iteration function here - in this case `.reduce(` - rather than using simple iteration:

```
const lines = blocklyCode.split("\n");

return lines.reduce((acc, line, idx) => {
    if (line.includes(BLOCKLY_HIGHLIGHT_PREFIX)) {
        return acc;
    }

    acc += line;
    if (idx+1 < lines.length) {
        acc += "\n";
    }

    return acc;
}, "");
```

...or something